src/momento/cache_client_async.py

```diff
@@ -770,6 +779,122 @@ async def set_remove_elements(
         """
         return await self._data_client.set_remove_elements(cache_name, set_name, elements)
 
+    async def sorted_set_put_element(
+        self,
+        cache_name: str,
+        sorted_set_name: str,
+        value: str | bytes,
+        score: float,
+        *,
+        ttl: CollectionTtl = CollectionTtl.from_cache_ttl(),
+    ) -> CacheSortedSetPutElementResponse:
+        """Sets a sorted set element to a value and score with a given time to live (TTL) seconds.
+
+        Args:
+            cache_name (str): Name of the cache containing the sorted set.
+            sorted_set_name (str): Name of the sorted set to set.
+            value (str | bytes): The value of the element to add.
+            score (float): The score of the element to add.
+            ttl: (CollectionTtl, optional): How to treat the sorted set's TTL.
+                Defaults to `CollectionTtl.from_cache_ttl()`
+
+        Returns:
+            CacheSortedSetPutElementResponse: result of the put operation.
+        """
+        put_elements_response = await self.sorted_set_put_elements(
+            cache_name, sorted_set_name, elements={value: score}, ttl=ttl
+        )
+        if isinstance(put_elements_response, CacheSortedSetPutElements.Success):
+            return CacheSortedSetPutElement.Success()
+        elif isinstance(put_elements_response, CacheSortedSetPutElements.Error):
+            return CacheSortedSetPutElement.Error(put_elements_response.inner_exception)
+        else:
+            return CacheSortedSetPutElement.Error(
+                UnknownException(f"Unknown put elements response: {put_elements_response}")
+            )
+
+    async def sorted_set_put_elements(
+        self,
+        cache_name: str,
+        sorted_set_name: str,
+        elements: TSortedSetElements,
+        *,
+        ttl: CollectionTtl = CollectionTtl.from_cache_ttl(),
+    ) -> CacheSortedSetPutElementsResponse:
+        """Puts elements into a sorted set.
+
+        Args:
+            cache_name (str): Name of the cache containing the sorted set.
+            sorted_set_name (str): The name of the sorted set to add to.
+            elements (TSortedSetElements): The elements to add.
+            ttl: (CollectionTtl, optional): How to treat the sorted set's TTL.
+                Defaults to `CollectionTtl.from_cache_ttl()`
+
+        Returns:
+            CacheSortedSetPutElementsResponse
+        """
+        return await self._data_client.sorted_set_put_elements(cache_name, sorted_set_name, elements, ttl)
+
+    async def sorted_set_fetch_by_score(
+        self,
+        cache_name: str,
+        sorted_set_name: str,
+        min_score: float | None = None,
+        max_score: float | None = None,
+        sort_order: SortOrder = SortOrder.ASCENDING,
+        offset: int | None = None,
+        count: int | None = None,
```

Comment:
The type hint for `type | None` is `Optional[type]`
```suggestion
        min_score: Optional[float] = None,
        max_score: Optional[float] = None,
        sort_order: SortOrder = SortOrder.ASCENDING,
        offset: Optional[int] = None,
        count: Optional[int] = None,
```

Reply:
The odd thing here in Python is that `Optional[float]` strictly means `float | None`, ie is nullable, and says nothing about whether it accepts a default value. Hence:

```python
def foo(x: Optional[int]) -> None
```

means `foo` is a void function that takes a single required argument `x`, which is a nullable int, vs

```python
def foo(x: Optional[int] = 42) -> None
```

means `foo` is a void function that takes an optional (ie not required) nullable int, which defaults to 42.

Reply:
I'll update the pr with optionals.